RFS/src/main/java/org/opensearch/migrations/bulkload/transformers/CompositeTransformer.java

```diff
@@ -19,10 +23,13 @@ public GlobalMetadata transformGlobalMetadata(GlobalMetadata globalData) {
     }
 
     @Override
-    public IndexMetadata transformIndexMetadata(IndexMetadata indexData) {
-        for (Transformer transformer : transformers) {
-            indexData = transformer.transformIndexMetadata(indexData);
-        }
-        return indexData;
+    public List<IndexMetadata> transformIndexMetadata(IndexMetadata indexData) {
+        return Stream.of(transformers)
+                .reduce(
+                    Stream.of(indexData),
+                    (stream, transformer) -> stream.flatMap(data -> transformer.transformIndexMetadata(data).stream()),
```

Comment:
The spirit of this call - slicing the output into separate subsequent transforms is really weird.  For the JSON Transformer composite transformer code, we'll do direct composition.  In this case, there's an implicit fanout because the type for the per-index metadata transform calls is a scalar Index - not a collection of Indices.

My opinion here, from the limited code that I've seen, would be to abstain from allowing the RFS-MetaData CompositeTransformer to host transforms that can return lists.  Keep two types of transforms.  Merge the outputs of the list one with a specific class that does fanout across another sequence of 'simple' transforms.  That might be a few more lines of code, but it might be a lot clearer.